File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {

        string r = "";

        cout << strs[0].size() << endl;

        sort(strs.begin(), strs.end());

        for (int i = 0; i < strs[0].size(); i++) {
            if (strs[0][i] == strs[strs.size() - 1][i]) {
                r += strs[0][i];
            }else{
                break;
            }
        }

        return r;
    }
};

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        result = ""
        for i in range(len(strs[0])):
            if strs[0][i] == strs[len(strs) - 1][i]:
                result += strs[0][i]
            else:
                return result

        return result # altho this is not required because it wont be reached